src/bootstrap/builder.rs

```diff
@@ -1140,6 +1140,14 @@ impl<'a> Builder<'a> {
             }
         }
 
+        // Compile everything except libraries and proc macros with the more
+        // efficient initial-exec TLS model. This doesn't work with `dlopen`,
+        // so we can't use it by default in general, but we can use it for tools
+        // and our own internal libraries.
+        if mode != Mode::Std && mode != Mode::Codegen {
```

Comment:
Might there be value in exhaustively matching the modes to ensure that we don't accidentally apply this flag to wrong things in the future, when somebody adds a new mode(s)? I could see it also making sense as a method along the lines of `Mode::is_dlopenable` or something...

Reply:
@nagisa I don't think it's worth exhaustively matching, but I do think factoring out a method makes sense. Done.